linkerd/app/src/admin.rs

```diff
@@ -60,10 +57,7 @@ impl Config {
             )
             .push_map_target(Target::from)
             .push(http::NewServeHttp::layer(Default::default(), drain.clone()))
-            .push_map_target(HttpAccept::from)
-            .push(svc::UnwrapOr::layer(
-                svc::Fail::<_, AdminHttpOnly>::default(),
-            ))
+            .push(svc::Filter::<TcpAccept, _>::layer(HttpAccept::try_from))
```

Comment:
nit/TIOLI, not a blocker for this PR: wonder if a `push_filter` would help reduce some boilerplate...there could even be a `push_try_from` adapter that just pushes `Filter<R::TryFrom>`...but that might not be inferrable...